x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/SchemaUtil.java

```diff
@@ -134,8 +137,7 @@ public static void getDestinationFieldMappings(final Client client,
             if (destinationMapping != null) {
```

Comment:
needs an additional `else if` to check for `dynamic`, would be good to do a `logger.info` for this case